### ISSUE TITLE
When pre_first_boot script Components Fail, workflow Execution Doesn't Stop

### ISSUE THREAD

**Issue by foigus** [grahamgilbert/imagr#51]
When running a workflow with a pre_first_boot script component, if the pre_first_boot script exits non-zero (i.e. it fails) workflow execution continues and the user isn't notified the script failed.

An example workflow:

```
<dict>
  <key>name</key>
  <string>Test Script Error</string>
  <key>description</key>
  <string>Intentionally create an error in a script</string>
  <key>components</key>
  <array>
    <dict>
      <key>type</key>
      <string>script</string>
      <key>content</key>
      <string>#!/bin/bash ; exit 1</string>
      <key>pre_first_boot</key>
      <true/>
    </dict>
  </array>
  <key>restart_action</key>
  <string>none</string>
</dict>
```

**Comment by grahamgilbert** [comment]
This should be rolled in with issue #48 - basically the same problem.

**Comment by grahamgilbert** [comment]
Fixed in #54
